tools/linker/CoreOptimizeGeneratedCode.cs

```diff
@@ -18,16 +19,31 @@
 
 namespace Xamarin.Linker {
 
+#if NET
```

Comment:
I don't want to block this PR but wouldn't be better to adopt this method to net6? Looking at the logic it seems to be a simpler version of dead blocks elimination which linker does automatically. At first look, all the code looks like it can be replaced with few substitutions.

Reply:
Yes, it might be possible to port it to use substitutions. Our list of things to do is pretty big though, and the current code isn't blocking anything.

Reply:
Also much easier to debug / compare by doing this step-by-step.

Reply:
I'm not sure substitutions, as available, today would work here. At least not for `IsDirectBinding` since the decision to always make it `true` is based on each individual type (computed in an earlier step) and not on some global state (known before build) given to the linker.

There are other places where substitutions would work (e.g. what remains of `RemoveCode`). We can start there and compare the results (since it won't be identical).